### PR TITLE
Bug 1255882: Add redirects from old or 301s file

### PR DESCRIPTION
**Pull request by pmac**

### CODE REVIEW

**Comment by pmac** [comment]
@jgmize or @glogiotatidis r?

File: bedrock/firefox/redirects.py [hunk 1]
```diff
@@ -317,6 +317,86 @@ def firefox_mobile_faq(request, *args, **kwargs):
...
+             'Archive/Mozilla/ActiveX_Control_for_Hosting_Netscape_Plug-ins_in_IE'),
+    redirect('^projects/plugins/xembed-plugin-extension\.html$',
+             'https://developer.mozilla.org/Add-ons/Plugins/XEmbed_Extension_for_Mozilla_Plugins'),
+    redirect('^projects/netlib/http/http-debugging\.html$',
+             'https://developer.mozilla.org/docs/Mozilla/Debugging/HTTP_logging'),
+    redirect('^projects/netlib/integrated-auth\.html$',
+             'https://developer.mozilla.org/docs/Mozilla/Integrated_authentication'),
+    redirect('^projects/netlib/Link_Prefetching_FAQ\.html$',
+             'https://developer.mozilla.org/docs/Web/HTTP/Link_prefetching_FAQ'),
+    redirect(r'^projects/embedding/GRE\.html', 'http://developer.mozilla.org/en/GRE'),
```
Comment (glogiotatidis) [inline bedrock/firefox/redirects.py]: Better `https://developer.mozilla.org/GRE`

File: bedrock/firefox/redirects.py [hunk 2]
```diff
@@ -317,6 +317,86 @@ def firefox_mobile_faq(request, *args, **kwargs):
...
+    redirect(r'^projects/embedding/windowAPIs\.html',
+             'http://developer.mozilla.org/en/Embedded_Dialog_API'),
```
Comment (glogiotatidis) [inline bedrock/firefox/redirects.py]: `s/http/https/` here and in following lines.

Comment (glogiotatidis) [inline bedrock/firefox/redirects.py]: Also remove `/en/` here and in following lines.

Comment (glogiotatidis) [inline bedrock/firefox/redirects.py]: Maybe even replace `/en/` with `/docs/` to help MDN de-pollute it's namespace in the future?

File: bedrock/firefox/redirects.py [hunk 3]
```diff
@@ -326,50 +406,87 @@ def firefox_mobile_faq(request, *args, **kwargs):
...
+    redirect(r'^projects/xslt(/.*)?$', 'https://developer.mozilla.org/docs/Web/XSLT'),
+    redirect(r'^projects/xul(/.*)?$', 'https://developer.mozilla.org/docs/Mozilla/Tech/XUL'),
     redirect(r'^quality/help(/.*)?$', 'http://quality.mozilla.org/get-involved'),
     redirect(r'^quality(/.*)?$', 'http://quality.mozilla.org/'),
 
     # Bug 654614 /blocklist -> addons.m.o/blocked
     redirect(r'^blocklist(/.*)?$', 'https://addons.mozilla.org/blocked/'),
 
+    redirect('^products/firebird$', '/firefox/products/'),
+    redirect('^products/firebird/download/$', '/firefox/new/'),
```
Comment (glogiotatidis) [inline bedrock/firefox/redirects.py]: Use `firefox.new` for consistency? Here and in following lines. Also for other internal redirects

File: bedrock/firefox/redirects.py [hunk 4]
```diff
@@ -326,50 +406,87 @@ def firefox_mobile_faq(request, *args, **kwargs):
...
+    redirect('^products/firefox/search', '/firefox/features/'),
+    redirect('^products/firefox/shelf\.html$', 'https://blog.mozilla.org/press/awards/'),
+    redirect('^products/firefox/smart-keywords\.html$',
+             'https://support.mozilla.org/en-US/kb/Smart+keywords'),
+    redirect('^products/firefox/support/$', 'https://support.mozilla.org/'),
+    redirect('^products/firefox/switch', '/firefox/new/'),
+    redirect('^products/firefox/system-requirements', '/firefox/system-requirements/'),
+    redirect('^products/firefox/tabbed-browsing', '/firefox/desktop/'),
+    redirect('^products/firefox/text-zoom\.html$',
+             'http://support.mozilla.com/en-US/kb/Text+Zoom'),
```
Comment (glogiotatidis) [inline bedrock/firefox/redirects.py]: `s/http/https/`

File: bedrock/firefox/redirects.py [hunk 5]
```diff
@@ -326,50 +406,87 @@ def firefox_mobile_faq(request, *args, **kwargs):
...
+    redirect('^products/firefox/themes$', 'https://addons.mozilla.org/themes/'),
+    redirect('^products/firefox/themes\.html$', 'https://addons.mozilla.org/themes/'),
+    redirect('^products/firefox/ui-customize\.html$',
+             'http://support.mozilla.com/en-US/kb/How+to+customize+the+toolbar'),
```
Comment (glogiotatidis) [inline bedrock/firefox/redirects.py]: `s/http/https/`

File: bedrock/firefox/redirects.py [hunk 6]
```diff
@@ -326,50 +406,87 @@ def firefox_mobile_faq(request, *args, **kwargs):
...
+    redirect('^products/firefox/upgrade', '/firefox/new/'),
+    redirect('^products/firefox/why/$', '/firefox/desktop/'),
 
     # bug 857246 redirect /products/firefox/start/  to start.mozilla.org
     redirect(r'^products/firefox/start/?$', 'http://start.mozilla.org'),
```
Comment (glogiotatidis) [inline bedrock/firefox/redirects.py]: s/http/https/

**Comment by glogiotatidis** [comment]
Just a few nits. Thanks for working on this @pmclanahan  ! :)

**Comment by pmac** [comment]
@glogiotatidis updated. Good catches. Thanks for the review.

**Comment by glogiotatidis** [comment]
There still a few internal redirects that use urls instead urlnames like `/firefox/all/` but I don't want to block on this. 

Also a few non-https redirects but not part of this PR. We can change to https in a new PR.

Thanks @pmac! r+ :beers:

**Pull request merged**